### PR TITLE
[16.0][MIG] base_custom_filter: Migration to 16.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
@qrtl

### CODE REVIEW

**Review by kikopeiro — APPROVED** [review]
Functional review / no code review. :+1:  thanks !

File: base_custom_filter/models/ir_filters_group.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    _name = "ir.filters.group"
+    _description = "Filters Group"
+    _order = "sequence, name, id"
+
+    def _selection_type(self):
+        return [("filter", "Filter"), ("groupby", "Group By")]
+
+    sequence = fields.Integer()
+    model_id = fields.Selection(
+        selection="_list_all_models", string="Model", required=True
```
Comment (yostashiro) [inline base_custom_filter/models/ir_filters_group.py:18]: Can we remove the method from this module and instead use the standard method by adding an import of `from odoo.addons.base.models.ir_filters import IrFilters`?

```suggestion
        selection="IrFilters._list_all_models", string="Model", required=True
```

**Review by yostashiro — APPROVED** [review]
Code review.

File: base_custom_filter/models/ir_filters_group.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        return super(IrFiltersGroup, self).unlink()
+
+    @api.model
+    def _list_all_models(self):
+        lang = self.env.lang or "en_US"
+        self._cr.execute(
+            "SELECT model, COALESCE(name->>%s, name->>'en_US') FROM ir_model ORDER BY 2",
+            [lang],
+        )
+        return self._cr.fetchall()
```
Comment (yostashiro) [inline base_custom_filter/models/ir_filters_group.py]:
```suggestion
        return self.env["ir.filters"]._list_all_models()
```

**Comment by victoralmau** [comment]
Please, cherry-pick https://github.com/OCA/server-ux/pull/657 to commit history

**Review by victoralmau — APPROVED** [review]
Code and functional review OK

**Pull request merged**